pkg/build/builder/common.go

```diff
@@ -66,7 +65,22 @@ func buildInfo(build *api.Build) []KeyValue {
 	return kv
 }
 
-func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *git.SourceInfo) {
+// updateBuildStatus updates the current status of the build only if there was
+// an error
+func updateBuildFailureReason(build *api.Build, reason api.StatusReason) {
+	switch reason {
+	case api.StatusReasonPostCommitHookFailed:
+		build.Status.Message = "Failed because of post commit hook"
+	case api.StatusReasonPushImageToRegistryFailed:
+		build.Status.Message = "Failed pushing the image to the registry"
```

Comment:
i didn't get there, yes. getting the errors out of s2i is going to be a major pain, if i remember correctly from what @php-coder told me. 
so that PR is inbound, when i figure out what/who/where